### PR TITLE
Adds callback for Long Press Up

### PR DESCRIPTION
**Pull request by jslavitz**
Fixes #18949 originally created by @alchalade

### CODE REVIEW

File: packages/flutter/lib/src/gestures/long_press.dart [hunk 1]
```diff
@@ -11,28 +11,50 @@ import 'recognizer.dart';
...
+    if (event is PointerUpEvent) {
+      if (_longPressAccepted == true && onLongPressUp != null) {
+        _longPressAccepted = false;
+        invokeCallback<void>('onLongPressUp', onLongPressUp);
+      } else {
+        resolve(GestureDisposition.rejected);
+      }
+    }
+
+    else if (event is PointerDownEvent) {
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/gestures/long_press.dart]: This should be directly after the closing `}` from the if statement above

File: packages/flutter/lib/src/gestures/long_press.dart [hunk 2]
```diff
@@ -11,28 +11,50 @@ import 'recognizer.dart';
...
   ///
   /// Consider assigning the [onLongPress] callback after creating this object.
-  LongPressGestureRecognizer({ Object debugOwner }) : super(deadline: kLongPressTimeout, debugOwner: debugOwner);
+  LongPressGestureRecognizer({ Object debugOwner })
+      : super(deadline: kLongPressTimeout, debugOwner: debugOwner);
+
+  bool _longPressAccepted = false;
 
-  /// Called when a long-press is recognized.
+  /// Called when the pointer stops contacting the screen after the long-press gesture has been recognized.
```
Comment (amirh) [inline packages/flutter/lib/src/gestures/long_press.dart]: Looks like this is the onLongPressUp comment?

File: packages/flutter/lib/src/gestures/long_press.dart [hunk 3]
```diff
@@ -11,28 +11,50 @@ import 'recognizer.dart';
...
   GestureLongPressCallback onLongPress;
 
+  /// Called when the pointer stops contacting the screen after the long-press gesture has been recognized
```
Comment (amirh) [inline packages/flutter/lib/src/gestures/long_press.dart]: nit: period

File: packages/flutter/lib/src/gestures/long_press.dart (hunk 1 above)
Comment (amirh) [inline packages/flutter/lib/src/gestures/long_press.dart]: should we set _longPressAccepted to null if the event is PointerCancelEvent?

File: packages/flutter/test/widgets/gesture_detector_test.dart [hunk 4]
```diff
@@ -314,4 +314,33 @@ void main() {
...
+      ),
+    );
+
+    Future<Null> longPress(Duration timeout) async {
+      final TestGesture gesture = await tester.startGesture(const Offset(400.0, 50.0));
+      await tester.pump(timeout);
+      await gesture.up();
+    }
+
+    await longPress(kLongPressTimeout * 1.5); // To make sure the time for long press has occured
```
Comment (amirh) [inline packages/flutter/test/widgets/gesture_detector_test.dart]: nit: can we tighten this to be `kLongPressTimrout + 1`

File: packages/flutter/lib/src/gestures/long_press.dart (hunk 1 above)
Comment (jslavitz) [reply packages/flutter/lib/src/gestures/long_press.dart]: I feel like that makes sense!

Comment (amirh) [reply packages/flutter/lib/src/gestures/long_press.dart]: oops I meant false not null

File: packages/flutter/lib/src/gestures/long_press.dart [hunk 5]
```diff
@@ -11,28 +11,50 @@ import 'recognizer.dart';
...
+        _longPressAccepted = false;
+        invokeCallback<void>('onLongPressUp', onLongPressUp);
+      } else {
+        resolve(GestureDisposition.rejected);
+      }
+    } else if (event is PointerDownEvent) {
+      // the first touch, initialize the  flag with false
+      _longPressAccepted = false;
+    } else if (event is PointerCancelEvent) {
+      _longPressAccepted = null;
```
Comment (amirh) [inline packages/flutter/lib/src/gestures/long_press.dart]: set this to false

Comment (amirh) [reply packages/flutter/lib/src/gestures/long_press.dart]: you can probably just make it
```dart
else if (event is PointerDownEvent || event is PointerCancelEvent( {
```

**Review by amirh — APPROVED** [review]
a few nits, but LGTM

**Pull request merged**